providers/git/src/airflow/providers/git/bundles/git.py

```diff
@@ -79,16 +80,20 @@ def __init__(
         )
 
         self._log.debug("bundle configured")
-        try:
-            self.hook = GitHook(git_conn_id=self.git_conn_id, repo_url=self.repo_url)
-            self.repo_url = self.hook.repo_url
-            self._log.debug("repo_url updated from hook", repo_url=self.repo_url)
-        except AirflowException as e:
-            self._log.warning("Could not create GitHook", conn_id=self.git_conn_id, exc=e)
+        self.hook: GitHook | None = None
+        if self.git_conn_id:
+            try:
+                self.hook = GitHook(git_conn_id=self.git_conn_id, repo_url=self.repo_url)
+            except AirflowException as e:
+                self._log.warning("Could not create GitHook", conn_id=self.git_conn_id, exc=e)
+            else:
+                self.repo_url = self.hook.repo_url
+                self._log.debug("repo_url updated from hook", repo_url=self.repo_url)
```

Comment:
I think we should not accept cases of both repo_url and git_conn_id.
From what I understand the repo_url  means that we ignore the git_conn_id.
So both parameters should be mutually exclusive
Similar to:
https://github.com/apache/airflow/blob/243fe86d4b3e59bb12977b3e36ca3f2ed27ca0f8/airflow-core/src/airflow/secrets/local_filesystem.py#L203-L207

This is important because consider a case where user provided both, then someone went to Connection->Git and changed the connection name. This code will automatically change to use the `repo_url` without the knowledge of the dag author. It could be something that he did not intent.

Reply:
The current behaviour is the other way around; `repo_url` is ignored if `git_conn_id` is provided.

If `repo_url` cannot be provided when loading from connection, that happens when a connection cannot be loaded? (The except block.)

Reply:
Regardless, why would you supply both if only one considered? This isn't a fallback case... Its a mutal exclusive case isn't it?

Reply:
Left a comment: https://github.com/apache/airflow/pull/49270#pullrequestreview-2767486702

Reply:
Left a comment: https://github.com/apache/airflow/pull/49270#pullrequestreview-2767486702

Reply:
In my first iteration of this, i implemented this order of precedence:
* if repo_url defined in both, use the on in the bundle confige. that seems right to me.

Reply:
the "why would you supply both" thing...

the issue i envision is you have one git conn that you want to use for multiple things.  

multiple bundles could use the same git conn, but not the other way around.

so therefore it seems reasonable to defer to the bundle config.

it's like when you have an operator, sometimes an option may be specified either at hook or operator, but usually if in both, you defer to operator, for the same reason.

Reply:
Oh interesting ----- if you define a repo url in gitdagbundle, then it is passed to githook and used as repo url _instead_ of taken from host in the git hook. 

so the precedence is as i indicated, it just was not obvious